maven-core/src/main/java/org/apache/maven/plugin/MavenPluginValidator.java

```diff
@@ -19,63 +19,16 @@
  * under the License.
  */
 
-import java.util.ArrayList;
 import java.util.List;
-
 import org.apache.maven.artifact.Artifact;
 import org.apache.maven.plugin.descriptor.PluginDescriptor;
 
 /**
  * MavenPluginValidator
  */
-public class MavenPluginValidator
+public interface MavenPluginValidator
 {
-    private final Artifact pluginArtifact;
-
-    private List<String> errors = new ArrayList<>();
-
-    private boolean firstDescriptor = true;
-
-    public MavenPluginValidator( Artifact pluginArtifact )
-    {
-        this.pluginArtifact = pluginArtifact;
-    }
-
-    public void validate( PluginDescriptor pluginDescriptor )
-    {
-        /*
-         * NOTE: For plugins that depend on other plugin artifacts the plugin realm contains more than one plugin
-         * descriptor. However, only the first descriptor is of interest.
-         */
-        if ( !firstDescriptor )
```

Comment:
How is this solved after removal?

Reply:
This code seems to be unused because before the MavenPluginValidator is created by each pluginDescriptor in DefaultMavenPluginManager: https://github.com/apache/maven/blob/master/maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultMavenPluginManager.java#L246
I don't think we need it.